src/Core/Geographic/Extent.js

```diff
@@ -296,14 +296,31 @@ Extent.prototype.offsetScale = function offsetScale(bbox) {
  * @param {type} bbox
  * @returns {Boolean}
  */
-Extent.prototype.intersect = function intersect(bbox) {
+Extent.prototype.intersects = function intersect(bbox) {
     const other = bbox.as(this.crs());
     return !(this.west() >= other.east(this._internalStorageUnit) ||
              this.east() <= other.west(this._internalStorageUnit) ||
              this.south() >= other.north(this._internalStorageUnit) ||
              this.north() <= other.south(this._internalStorageUnit));
 };
 
+/**
+ * @documentation: Return the intersection of this extent with another one
+ * @param {type} other
+ * @returns {Boolean}
+ */
+Extent.prototype.intersect = function intersect(other) {
+    if (!this.intersects(other)) {
+        return new Extent(this.crs(), 0, 0, 0, 0);
```

Comment:
Do we have a `Extent.isEmpty` method? Or maybe we could return the same constant every time `EMPTY_EXTENT`, allowing client code to test `extent === EMPTY_EXTENT`? Feel free to skip this if you think nobody'll use it though.

Reply:
I have no strong feelings about this one... so I'd suggest to keep the code as is, and see if it becomes an issue?

Reply:
Fine by me